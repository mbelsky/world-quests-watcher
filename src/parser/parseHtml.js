import cheerio from "cheerio";

const s = {
  questsCount:
    ".listview-band-top > .listview-nav.listview-nav-nocontrols > span > b",
  row: ".listview-row",
  rowData: "td",
};

export function parseHtml(html) {
  const $ = cheerio.load(html);
  const questsCountNodes = $(s.questsCount);
  const nodeCounts = questsCountNodes.length;

  if (nodeCounts !== 3) {
    return {
      error: new Error(
        `Invalid questsCountNodes.length=${nodeCounts}. html: ${html}`
      ),
    };
  }

  const totalQuestsCount = questsCountNodes.last().text();

  const rowNodes = $(s.row);
  const rowNodesCount = rowNodes.length;
  const numberTotalQuestsCount = Number(totalQuestsCount);

  if (rowNodesCount !== numberTotalQuestsCount) {
    return {
      error: new Error(
        `rowNodesCount=${rowNodesCount} !== totalQuestsCount=${numberTotalQuestsCount}. html: ${html}`
      ),
    };
  }

  function mapRowNodes(_, rowNode) {
    rowNode = $(rowNode);

    const dataList = $(s.rowData, rowNode);

    if (dataList.length !== 5) {
      throw new Error(`Invalid dataList.length=${dataList.length}`);
    }

    const href = dataList.find("a").attr("href");
    const endsAt = dataList.eq(3).find("div").text().trim();
    const id = href.split('=')[1]

    return { endsAt, href, id };
  }

  try {
    const quests = rowNodes.map(mapRowNodes).get();

    return {
      data: quests,
    };
  } catch (e) {
    return {
      error: new Error(e.message + ". html: " + html),
    };
  }
}
