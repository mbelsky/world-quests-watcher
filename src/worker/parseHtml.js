import cheerio from "cheerio";

const s = {
  questsCount:
    ".listview-band-top > .listview-nav.listview-nav-nocontrols > span > b",
  row: ".listview-row",
  rowData: "td",
};

export function parseHtml(html) {
  const $ = cheerio.load(html);
  const questsCountNodes = $(s.questsCount);
  const nodeCounts = questsCountNodes.length;

  if (nodeCounts !== 3) {
    return {
      error: new Error(
        `Invalid questsCountNodes.length=${nodeCounts}. html: ${html}`
      ),
    };
  }

  const totalQuestsCount = questsCountNodes.last().text();

  const rowNodes = $(s.row);
  const rowNodesCount = rowNodes.length;
  const numberTotalQuestsCount = Number(totalQuestsCount);

  if (rowNodesCount !== numberTotalQuestsCount) {
    return {
      error: new Error(
        `rowNodesCount=${rowNodesCount} !== totalQuestsCount=${numberTotalQuestsCount}. html: ${html}`
      ),
    };
  }

  function mapRowNodes(_, rowNode) {
    rowNode = $(rowNode);

    const dataList = $(s.rowData, rowNode);

    if (dataList.length !== 5) {
      throw new Error(`Invalid dataList.length=${dataList.length}`);
    }

    const questAnchor = dataList.eq(0).find("a");
    const href = questAnchor.attr("href");
    const name = questAnchor.text().replace(/\s+/g, " ");
    const endsAt = dataList.eq(3).find("div").text().trim();
    const [__, id] = href.split("=");

    return { endsAt, href, id, name };
  }

  try {
    const quests = rowNodes.map(mapRowNodes).get();

    return {
      data: quests,
    };
  } catch (e) {
    return {
      error: new Error(e.message + ". html: " + html),
    };
  }
}

export function parseHtmlMap(htmlMap) {
  const questsMap = {};

  Object.entries(htmlMap).forEach(([key, html]) => {
    const [region] = key.split("-");

    const { data, error } = parseHtml(html);

    if (error) {
      // ğŸ˜­
      throw error;
    }

    const quests = data.map((quest) => {
      return {
        ...quest,
        region,
      };
    });

    if (region in questsMap) {
      questsMap[region].push(...quests);
    } else {
      questsMap[region] = quests;
    }
  });

  return questsMap;
}
