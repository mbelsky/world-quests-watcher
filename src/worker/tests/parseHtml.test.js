import { parseHtml, parseHtmlMap } from "../parseHtml.js";
import { html, htmlMap } from "./input.mock.js";

test("parseHtml", () => {
  const result = parseHtml(html);

  expect(result).toMatchSnapshot();
});

test("parseHtmlMap", () => {
  const result = parseHtmlMap(htmlMap);

  expect(result.eu.length).toBe(8)
  expect(result.na.length).toBe(8)
  expect(result).toMatchSnapshot();
});
