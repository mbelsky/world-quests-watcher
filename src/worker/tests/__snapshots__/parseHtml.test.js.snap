// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseHtml 1`] = `
Object {
  "data": Array [
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=42119",
      "id": "42119",
      "name": "A Cleansing Cocktail",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42859",
      "id": "42859",
      "name": "A Draught of Hope",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43964",
      "id": "43964",
      "name": "A Jarl's Feast",
    },
    Object {
      "endsAt": "15 hours 1 minute",
      "href": "/quest=46932",
      "id": "46932",
      "name": "A Tad More Corruption",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=41705",
      "id": "41705",
      "name": "Assault on Thunder Totem",
    },
    Object {
      "endsAt": "20 hours 1 minute",
      "href": "/quest=47705",
      "id": "47705",
      "name": "Behind Legion Lines",
    },
    Object {
      "endsAt": "3 days 10 hours",
      "href": "/quest=43642",
      "id": "43642",
      "name": "Black Rook Hold: The Sorrow",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=47561",
      "id": "47561",
      "name": "Blistermaw",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41308",
      "id": "41308",
      "name": "Border Reinforcements",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41209",
      "id": "41209",
      "name": "Brimstone Destroyer",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=42151",
      "id": "42151",
      "name": "Clear the Skies",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=48509",
      "id": "48509",
      "name": "Commander Sathrenael",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=48510",
      "id": "48510",
      "name": "Commander Vecaya",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=46168",
      "id": "46168",
      "name": "Commander Vorlax",
    },
    Object {
      "endsAt": "20 hours 1 minute",
      "href": "/quest=46821",
      "id": "46821",
      "name": "Creepy Crawlies",
    },
    Object {
      "endsAt": "15 hours 1 minute",
      "href": "/quest=48286",
      "id": "48286",
      "name": "Crystal Methods",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44856",
      "id": "44856",
      "name": "Cut the Cattitude",
    },
    Object {
      "endsAt": "1 day 10 hours",
      "href": "/quest=44189",
      "id": "44189",
      "name": "DANGER: Bestrix",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42785",
      "id": "42785",
      "name": "DANGER: Den Mother Ylva",
    },
    Object {
      "endsAt": "5 days 10 hours",
      "href": "/quest=43798",
      "id": "43798",
      "name": "DANGER: Kosumoth the Hungering",
    },
    Object {
      "endsAt": "2 days 10 hours",
      "href": "/quest=42927",
      "id": "42927",
      "name": "DANGER: Malisandra",
    },
    Object {
      "endsAt": "1 day 22 hours",
      "href": "/quest=44893",
      "id": "44893",
      "name": "Direbeak Swarm!",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=42276",
      "id": "42276",
      "name": "Disgusting, but Useful",
    },
    Object {
      "endsAt": "1 day 4 hours",
      "href": "/quest=43772",
      "id": "43772",
      "name": "Enigmatic",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=47720",
      "id": "47720",
      "name": "Eternal Vengeance",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=48722",
      "id": "48722",
      "name": "Feasel the Muffin Thief",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=46077",
      "id": "46077",
      "name": "Felbound Spirits",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41560",
      "id": "41560",
      "name": "Felhide",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41511",
      "id": "41511",
      "name": "Felwort",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=45934",
      "id": "45934",
      "name": "Fowl Play",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=41014",
      "id": "41014",
      "name": "Fury Road",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48097",
      "id": "48097",
      "name": "Gatekeeper's Review: Cunning",
    },
    Object {
      "endsAt": "1 day 10 hours",
      "href": "/quest=46139",
      "id": "46139",
      "name": "Gemcutter Needed",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44746",
      "id": "44746",
      "name": "Helarjar Landing: Rockaway Coast",
    },
    Object {
      "endsAt": "15 hours 1 minute",
      "href": "/quest=48099",
      "id": "48099",
      "name": "Hostile Echology",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=46111",
      "id": "46111",
      "name": "Illidari Masters: Sissix",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44805",
      "id": "44805",
      "name": "In the Thick of It",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41944",
      "id": "41944",
      "name": "Jarrun's Ladder",
    },
    Object {
      "endsAt": "5 hours 1 minute",
      "href": "/quest=48386",
      "id": "48386",
      "name": "Jed'hin Tournament",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=42150",
      "id": "42150",
      "name": "Kal'delar Naga Incursion",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=46167",
      "id": "46167",
      "name": "Kozrum",
    },
    Object {
      "endsAt": "2 days 16 hours",
      "href": "/quest=43769",
      "id": "43769",
      "name": "Ley Race",
    },
    Object {
      "endsAt": "1 day 22 hours",
      "href": "/quest=43764",
      "id": "43764",
      "name": "Ley Race",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=43583",
      "id": "43583",
      "name": "Life Finds a Way",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=41602",
      "id": "41602",
      "name": "Lively Stormrays",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44786",
      "id": "44786",
      "name": "Midterm: Rune Aptitude",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=47552",
      "id": "47552",
      "name": "Mistress Il'thendra",
    },
    Object {
      "endsAt": "1 day 10 hours",
      "href": "/quest=41865",
      "id": "41865",
      "name": "Neltharion's Lair: Mother of Stone",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=45792",
      "id": "45792",
      "name": "Occultist Onslaught",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=48735",
      "id": "48735",
      "name": "Overseer Y'Sorna",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=49056",
      "id": "49056",
      "name": "Pilfer",
    },
    Object {
      "endsAt": "5 days 10 hours",
      "href": "/quest=42819",
      "id": "42819",
      "name": "Pocket Wizard",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=41964",
      "id": "41964",
      "name": "Poisoned Waters",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=40850",
      "id": "40850",
      "name": "Prisoners of Greystone",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=43324",
      "id": "43324",
      "name": "Rage of the Owlbeasts",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=46076",
      "id": "46076",
      "name": "Restless Spirits",
    },
    Object {
      "endsAt": "20 hours 1 minute",
      "href": "/quest=41926",
      "id": "41926",
      "name": "Returning Champion",
    },
    Object {
      "endsAt": "15 hours 1 minute",
      "href": "/quest=42176",
      "id": "42176",
      "name": "Rise of Skovald",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48833",
      "id": "48833",
      "name": "Ritual Completion",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48526",
      "id": "48526",
      "name": "Scuttled",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43709",
      "id": "43709",
      "name": "Seal Clubbing",
    },
    Object {
      "endsAt": "5 days 10 hours",
      "href": "/quest=48959",
      "id": "48959",
      "name": "Seat of the Triumvirate: Dark Fissures",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44815",
      "id": "44815",
      "name": "Sick of the Sycophants",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=47542",
      "id": "47542",
      "name": "Siegemaster Voraan",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41549",
      "id": "41549",
      "name": "Slab of Bacon",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48514",
      "id": "48514",
      "name": "Slumbering Behemoths",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44892",
      "id": "44892",
      "name": "Snowfeather Swarm!",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41343",
      "id": "41343",
      "name": "Solid Crabshell Fragment",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48338",
      "id": "48338",
      "name": "Supplies Needed: Astral Glory",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48374",
      "id": "48374",
      "name": "Supplies Needed: Lightweave Cloth",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=48192",
      "id": "48192",
      "name": "Tar Spitter",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=46170",
      "id": "46170",
      "name": "Thar'gokk",
    },
    Object {
      "endsAt": "1 day 10 hours",
      "href": "/quest=43637",
      "id": "43637",
      "name": "The Arcway: Clogged Drain",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=46146",
      "id": "46146",
      "name": "The Burning Shores",
    },
    Object {
      "endsAt": "5 hours 1 minute",
      "href": "/quest=45035",
      "id": "45035",
      "name": "The Call of Battle",
    },
    Object {
      "endsAt": "20 hours 1 minute",
      "href": "/quest=47712",
      "id": "47712",
      "name": "The Devastator",
    },
    Object {
      "endsAt": "2 days 10 hours",
      "href": "/quest=42170",
      "id": "42170",
      "name": "The Dreamweavers",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43755",
      "id": "43755",
      "name": "The Magic of Flight",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42082",
      "id": "42082",
      "name": "The Shattered Locus",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=45134",
      "id": "45134",
      "name": "The Soul Harvesters",
    },
    Object {
      "endsAt": "1 day 10 hours",
      "href": "/quest=42234",
      "id": "42234",
      "name": "The Valarjar",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42422",
      "id": "42422",
      "name": "The Wardens",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=40280",
      "id": "40280",
      "name": "Training with Bredda",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42159",
      "id": "42159",
      "name": "Training with the Nightwatchers",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=45379",
      "id": "45379",
      "name": "Treasure Master Iks'reeged",
    },
    Object {
      "endsAt": "5 hours 1 minute",
      "href": "/quest=41930",
      "id": "41930",
      "name": "Valor Calls",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=48465",
      "id": "48465",
      "name": "Vrax'thul",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44289",
      "id": "44289",
      "name": "WANTED: Arru",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=43622",
      "id": "43622",
      "name": "WANTED: Glimar Ironfist",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=43611",
      "id": "43611",
      "name": "WANTED: Inquisitor Tivos",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=44023",
      "id": "44023",
      "name": "WANTED: Siegemaster Aedrin",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=43451",
      "id": "43451",
      "name": "WANTED: Urgev the Flayer",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43599",
      "id": "43599",
      "name": "Warden Tower Assault: Blackhawk's Bulwark",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41257",
      "id": "41257",
      "name": "Warden Tower Assault: Cordana's Apex",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43248",
      "id": "43248",
      "name": "Warden Tower Assault: Darkfollow's Spire",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41420",
      "id": "41420",
      "name": "Warden Tower Assault: Nightwatcher's Perch",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=42070",
      "id": "42070",
      "name": "Warden Tower Assault: Starstalker's Point",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43600",
      "id": "43600",
      "name": "Warden Tower Assault: Whisperwind's Citadel",
    },
    Object {
      "endsAt": "1 hour 1 minute",
      "href": "/quest=46263",
      "id": "46263",
      "name": "Weaving Fel Webs",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=43943",
      "id": "43943",
      "name": "Withered Army Training",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=41677",
      "id": "41677",
      "name": "Work Order: Auto-Hammer",
    },
    Object {
      "endsAt": "16 hours 1 minute",
      "href": "/quest=41676",
      "id": "41676",
      "name": "Work Order: Pump-Action Bandage Gun",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=41647",
      "id": "41647",
      "name": "Work Order: Silkweave Robe",
    },
    Object {
      "endsAt": "4 hours 1 minute",
      "href": "/quest=41660",
      "id": "41660",
      "name": "Work Order: Skaggldrynk",
    },
    Object {
      "endsAt": "22 hours 1 minute",
      "href": "/quest=48696",
      "id": "48696",
      "name": "Wrangler Kravos",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48829",
      "id": "48829",
      "name": "Wrath-Lord Yarez",
    },
    Object {
      "endsAt": "10 hours 1 minute",
      "href": "/quest=48738",
      "id": "48738",
      "name": "Zul'tan the Numerous",
    },
  ],
}
`;

exports[`parseHtmlMap 1`] = `
Object {
  "eu": Array [
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=52986",
      "id": "52986",
      "name": "A Wicked Vessel",
      "region": "eu",
    },
    Object {
      "endsAt": "4 days 11 hours",
      "href": "/quest=57157",
      "id": "57157",
      "name": "Assault: The Black Empire",
      "region": "eu",
    },
    Object {
      "endsAt": "19 hours 54 minutes",
      "href": "/quest=50296",
      "id": "50296",
      "name": "Billy Goat Barber",
      "region": "eu",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=58747",
      "id": "58747",
      "name": "Brain Tickling",
      "region": "eu",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=52986",
      "id": "52986",
      "name": "A Wicked Vessel",
      "region": "eu",
    },
    Object {
      "endsAt": "4 days 11 hours",
      "href": "/quest=57157",
      "id": "57157",
      "name": "Assault: The Black Empire",
      "region": "eu",
    },
    Object {
      "endsAt": "19 hours 54 minutes",
      "href": "/quest=50296",
      "id": "50296",
      "name": "Billy Goat Barber",
      "region": "eu",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=58747",
      "id": "58747",
      "name": "Brain Tickling",
      "region": "eu",
    },
  ],
  "na": Array [
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=52986",
      "id": "52986",
      "name": "A Wicked Vessel",
      "region": "na",
    },
    Object {
      "endsAt": "4 days 11 hours",
      "href": "/quest=57157",
      "id": "57157",
      "name": "Assault: The Black Empire",
      "region": "na",
    },
    Object {
      "endsAt": "19 hours 54 minutes",
      "href": "/quest=50296",
      "id": "50296",
      "name": "Billy Goat Barber",
      "region": "na",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=58747",
      "id": "58747",
      "name": "Brain Tickling",
      "region": "na",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=52986",
      "id": "52986",
      "name": "A Wicked Vessel",
      "region": "na",
    },
    Object {
      "endsAt": "4 days 11 hours",
      "href": "/quest=57157",
      "id": "57157",
      "name": "Assault: The Black Empire",
      "region": "na",
    },
    Object {
      "endsAt": "19 hours 54 minutes",
      "href": "/quest=50296",
      "id": "50296",
      "name": "Billy Goat Barber",
      "region": "na",
    },
    Object {
      "endsAt": "11 hours 54 minutes",
      "href": "/quest=58747",
      "id": "58747",
      "name": "Brain Tickling",
      "region": "na",
    },
  ],
}
`;
